### DISCUSSION TITLE
Add Lucky::Params#[] methods

### DISCUSSION THREAD

**Discussion by grepsedawk** [luckyframework/lucky#1792] [Feature requests & Ideas]
It's always a bit of a shock to try params["param_name"] and get method undefined

Similar to https://github.com/luckyframework/lucky/pull/1678, I would love it if we added `[]` methods to params

**Comment by jwoertink** [comment]
We can't really do this without forcing all `Avram::Paramable` objects to start acting like a hash. It would start throwing things off with all of the nested values because then we'd have to also support things like `params["user"]["username"]`, or `params["posts"][0]["title"]`. 

What I do is just keep in mind that it's not a hash-like object. Maybe we just add a compile-time error when you try to use `[]` that says to use `get`, or links to docs or something?

**Reply by watzon** [reply]
The more things we can warn about at compile time the better in my opinion

**Reply by grepsedawk** [reply]
`[]` is just VERY much more pretty, and saves 5 characters in line length.
Even if it was just available at the root level `params` I still feel adding `[]` provides a ton of value here.

At the end of the day, `[]` doesn't mean "hash like", it's crystal's way of accepting keys and returning values, which is precisely what params does.

**Reply by grepsedawk** [reply]
`[]` is also intuitive and expected
